Generator.Equals/EqualsGenerator.cs

```diff
@@ -20,103 +21,67 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             var provider = context.SyntaxProvider
                 .ForAttributeWithMetadataName(
                     "Generator.Equals.EquatableAttribute",
-                    (syntaxNode, _) =>
-                    {
-                        return syntaxNode switch
-                        {
-                            ClassDeclarationSyntax => true,
-                            RecordDeclarationSyntax => true,
-                            StructDeclarationSyntax => true,
-                            _ => false
-                        };
-                    },
+                    (syntaxNode, _) => syntaxNode is ClassDeclarationSyntax or RecordDeclarationSyntax or StructDeclarationSyntax,
                     (syntaxContext, ct) => syntaxContext);
```

Comment:
Don't return the `syntaxContext`. Instead transform it _here_ into a cacheable value, namely the `record`s that you have already defined elsewhere.

Returning the `syntaxContext` removes the ability to cache the output, which means that you no longer have an incremental generator. _All_ usages of the attribute will rerun the entire pipeline on every keystroke.

Using a cacheable value means that only the one that actually changes will need to be generated, reducing memory usage and improving performance.